aptos-move/aptos-release-builder/src/simulate.rs

```diff
@@ -0,0 +1,596 @@
+            name: Identifier::new(GasScheduleV2::TYPE_IDENTIFIER).unwrap(),
+            type_args: vec![],
+        })?,
+        StateValue::new_legacy(
+            bcs::to_bytes(&GasScheduleV2 {
+                feature_version: gas_feature_version,
+                entries: gas_params.to_on_chain_gas_schedule(gas_feature_version),
+            })?
+            .into(),
+        ),
+    );
+
+    // Execute the governance scripts in sorted order.
+    println!("Executing governance scripts...");
+
+    const DUMMY_PROPOSAL_ID: u64 = u64::MAX;
+
+    for (script_idx, (script_path, script_blob)) in scripts.iter().zip(compiled_scripts).enumerate()
+    {
+        // Patch framework functions to skip the governance process.
+        // This is redone every time we execute a script because the previous script could have
+        // overwritten the framework.
+        let features = Features::fetch_config(&state_view)
+            .ok_or_else(|| anyhow!("failed to fetch feature flags"))?;
+        let deserializer_config = aptos_prod_deserializer_config(&features);
+
+        patch_create_signer(&state_view, &deserializer_config)?;
+        // If the script is the last step of the proposal, it MUST NOT have a next execution hash.
+        // Set the boolean flag to true to use a modified patch to catch this.
+        let forbid_next_execution_hash = script_idx == scripts.len() - 1;
+        patch_resolve_multistep_proposal(
+            &state_view,
+            &deserializer_config,
+            forbid_next_execution_hash,
+        )?;
+
+        let script_name = script_path.file_name().unwrap().to_string_lossy();
+        println!("    {}", script_name);
+
+        // Create a new VM to ensure the loader is clean.
+        // The warm vm cache also needs to be explicitly flushed as it cannot detect the
+        // patches we performed.
+        flush_warm_vm_cache();
+        let vm = AptosVM::new(&state_view);
+        let log_context = AdapterLogSchema::new(state_view.id(), 0);
+        let resolver = state_view.as_move_resolver();
+        let (_vm_status, vm_output) = vm.execute_user_transaction(
+            &resolver,
+            &account
+                .account()
+                .transaction()
+                .script(Script::new(script_blob, vec![], vec![
+                    TransactionArgument::U64(DUMMY_PROPOSAL_ID), // dummy proposal id, ignored by the patched function
+                ]))
+                .chain_id(chain_id.chain_id())
+                .sequence_number(script_idx as u64)
+                .gas_unit_price(gas_params.vm.txn.min_price_per_gas_unit.into())
+                .max_gas_amount(100000)
+                .ttl(u64::MAX)
+                .sign(),
+            &log_context,
+        );
+        // We require all governance scripts to trigger reconfiguration so check it here.
+        if AptosVM::should_restart_execution(vm_output.events()) {
```

Comment:
Should it be the negation of this?

Reply:
Good catch.

However looks like this would not be an easy fix. With dkg, reconfiguration is started by the script but may not actually happen until the next epoch. 

I guess I'll have to remove this check for now, and we'll need to discuss what the proper solution might be. I'm thinking that maybe `reconfiguration_with_dkg::try_start` should emit its own event indicating this.